modules/upgrade/hub.go

```diff
@@ -0,0 +1,122 @@
+package upgrade
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/blang/semver"
+)
+
+const (
+	// HubBaseURL base hub url
+	HubBaseURL = "https://hub.grid.tf/"
+
+	// HubStorage default hub db
+	HubStorage = "zdb://hub.grid.tf:9900"
+)
+
+// Hub API for f-list
+type Hub struct{}
+
+// MountURL returns the full url of given flist.
+func (h *Hub) MountURL(flist string) string {
+	url, err := url.Parse(HubBaseURL)
+	if err != nil {
+		panic("invalid base url")
+	}
+	url.Path = flist
+	return url.String()
+}
+
+// StorageURL return hub storage url
+func (h *Hub) StorageURL() string {
+	return HubStorage
+}
+
+// Info gets flist info from hub
+func (h *Hub) Info(flist string) (info FListInfo, err error) {
+	u, err := url.Parse(HubBaseURL)
+	if err != nil {
+		panic("invalid base url")
```

Comment:
is this valid to panic here ? why not returning an error ?

Reply:
Same answer as above.